marlowe-playground-client/src/Marlowe/Holes.purs

```diff
@@ -126,78 +129,96 @@ data Argument
   | NewtypeArg
   | GenArg MarloweType
 
-getMarloweConstructors :: MarloweType -> Map String (Array Argument)
-getMarloweConstructors ChoiceIdType = Map.singleton "ChoiceId" [ DefaultString "choiceNumber", DataArg PartyType ]
+data TermGenerator
+  = ArgumentArray (Array Argument)
+  | SimpleArgument Argument
+
+-- Shorter version of constructors for keeping code short
+aa :: Array Argument -> TermGenerator
```

Comment:
I think I prefer the longer names

Reply:
would a `Monoid` make sense here perhaps?

Reply:
in fact, I don't really get how `SimpleArgument` helps over just having a singleton list?

Reply:
`SimpleArgument` doesn't print the constructor, that is the difference between singleton and `SimpleArgument`. I am not sure what the original `Argument` does, it is just a patch so that it doesn't print the argument. I can put that in a comment. I guess that means `Monoid` doesn't make sense either